pipeline/import/taxaImport.R

```diff
@@ -41,19 +41,24 @@ regionGeometry <- defineRegion(level, region)
 if(file.exists(paste0(folderName, "/focalTaxa.csv"))){
   focalTaxon <- read.csv(paste0(folderName, "/focalTaxa.csv"), header = T)
 } else {
-  focalTaxon <- read.csv(paste0("data/external/focalTaxa.csv"), header = T)
-  # filter selected taxa
-  focalTaxon <- focalTaxon[focalTaxon$include,]
-  
-  # get missing keys
-  missingKey <- is.na(focalTaxon$key)
-  focalTaxon$key[missingKey] <- getUsageKeys(focalTaxon$scientificName[missingKey], 
-                                             rank = focalTaxon$level[missingKey], 
-                                             strict = T)
+  focalTaxon <- read.csv("data/external/focalTaxa.csv", header = T)
   # save for reference
   write.csv(focalTaxon, paste0(folderName, "/focalTaxa.csv"), row.names = FALSE)
 }
 
+# Refine focal taxon
+focalTaxon <- focalTaxon[focalTaxon$include,]
+
+# get missing keys
+missingKey <- is.na(focalTaxon$key)
+focalTaxon$key[missingKey] <- getUsageKeys(focalTaxon$scientificName[missingKey], 
```

Comment:
Could key assignment come before saving to folderName? when focalTaxa.csv is loaded in future scripts, they would lose the key

Reply:
Have just pushed a commit to solve this